microbench/src/main/java/io/netty/buffer/ByteBufAccessBenchmark.java

```diff
@@ -0,0 +1,150 @@
+            super(Unpooled.EMPTY_BUFFER);
+            this.byteBuffer = byteBuffer;
+        }
+        @Override
+        public ByteBuf setLong(int index, long value) {
+            byteBuffer.putLong(index, value);
+            return this;
+        }
+        @Override
+        public long getLong(int index) {
+            return byteBuffer.getLong(index);
+        }
+        @Override
+        public byte readByte() {
+            return byteBuffer.get();
+        }
+        @Override
+        public ByteBuf touch() {
+            // hack since WrappedByteBuf.readerIndex(int) is final
+            byteBuffer.position(0);
+            return this;
+        }
+        @Override
+        public boolean release() {
+            PlatformDependent.freeDirectBuffer(byteBuffer);
+            return true;
+        }
+    }
+
+    public enum ByteBufType {
+        UNSAFE {
+            @Override
+            ByteBuf newBuffer() {
+                return new UnpooledUnsafeDirectByteBuf(
+                        UnpooledByteBufAllocator.DEFAULT, 64, 64).setIndex(0, 64);
+            }
+        },
+        UNSAFE_SLICE {
+            @Override
+            ByteBuf newBuffer() {
+                return UNSAFE.newBuffer().slice(16, 48);
+            }
+        },
+        HEAP {
+            @Override
+            ByteBuf newBuffer() {
+                return new UnpooledUnsafeHeapByteBuf(
+                        UnpooledByteBufAllocator.DEFAULT, 64, 64).setIndex(0,  64);
+            }
+        },
+        COMPOSITE {
+            @Override
+            ByteBuf newBuffer() {
+                return Unpooled.wrappedBuffer(UNSAFE.newBuffer(), HEAP.newBuffer());
+            }
+        },
+        NIO {
+            @Override
+            ByteBuf newBuffer() {
+                return new NioFacade(ByteBuffer.allocateDirect(64));
+            }
+        };
+        abstract ByteBuf newBuffer();
+    }
+
+    @Param
+    public ByteBufType bufferType;
+
+    private ByteBuf buffer;
+
+    @Setup
+    public void setup() {
+        buffer = bufferType.newBuffer();
+    }
+
+    @TearDown
+    public void tearDown() {
+        buffer.release();
+    }
+
+    @Benchmark
+    public long setGetLong() {
+        return buffer.setLong(0, 1).getLong(0);
+    }
+
+    @Benchmark
+    public ByteBuf setLong() {
+        return buffer.setLong(0, 1);
+    }
+
+    @Benchmark
+    public void readEightBytes(final Blackhole bh) {
+        buffer.readerIndex(0).touch();
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+        bh.consume(buffer.readByte());
+    }
+
+    // doesn't seem to be possible to parameterize JVM args with JMH right now
+//  @Test
+//  public void runWithNoCheck() throws Exception {
```

Comment:
You can only be override:

```java
protected String[] jvmArgs() {
    return jvmArgs;
}
```

Reply:
This applies to the entire run though, can't differ based on parameters.

Reply:
`System.setProperty("io.netty.buffer.checkAccessible", "false")` could be placed on `@Setup` before accessing any buffer class and it should allow setting it per fork.

Reply:
Good idea!